packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -138,7 +155,18 @@ export const plugin = <TContext>(
       let didResolveSource: boolean = false;
 
       return {
+        didResolveOperation(requestContext) {
+          if (typeof options.traceReporting === 'function') {
+            options.traceReporting(requestContext).then(shouldReportTrace => {
+              if (!shouldReportTrace) didEnd(requestContext, false);
```

Comment:
TypeScript will prevent consumers from mistakes here, but JavaScript users might unwittingly return other values.  Might we be more explicit about the return value and check if it's specifically a `boolean`?  We could opt to not trace in that case and a `logger.warn` could (really) help a user debug a mistake here in the event that a JS user returned, e.g., a `Function` or `Objeect` (both truthy!).

TypeScript is still not always used, even if I wish it was!